### PR TITLE
Add option to convert to unscanned ckpt

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Create a new [PR](https://github.com/AI-Hypercomputer/maxtext/pull/1485) from old one to have a clean commit history.

# Tests

Run conversion for V3 and V2-Lite:
* V3 tested in MMLU scores 
* V2-Lite tested with same decoding results `Input I love to play ->  with my food. I love to make it` with ckpt converted from scanned version

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: MaxText/llama_or_mistral_ckpt.py [hunk 1]
```diff
@@ -957,7 +957,7 @@ def convert_to_jax_weights(base_model_path, model_size, huggingface_ckpt):
   return _convert_pytorch_to_jax_weights(base_model_path, model_size, model_params, mem_info)
 
 
-def save_weights_to_checkpoint(maxtext_model_path, jax_weights, device_count):
+def save_weights_to_checkpoint(maxtext_model_path, jax_weights, device_count, use_ocdbt=True, use_zarr3=True):
```
Comment (gagika) [inline MaxText/llama_or_mistral_ckpt.py:960]: is this intentional, it makes checkpoint with many files, right?

Perhaps we can pass those with command line arguments with default values?

**Review by gagika — APPROVED** [review]

File: MaxText/llama_or_mistral_ckpt.py (hunk 1 above)
Comment (RissyRan) [reply MaxText/llama_or_mistral_ckpt.py]: Updated.

**Review by mitalisi — APPROVED** [review]
Please add a TODO fo logits test. We can address in future PRs.

**Comment by RissyRan** [comment]
> Please add a TODO fo logits test. We can address in future PRs.

Sounds good. Added `TODO(ranran): add forward_pass_logit_chekcker.py test` into script for v3.
